.github/workflows/build-test-publish.yml

```diff
@@ -602,19 +617,19 @@ jobs:
       - name: start deployment
         uses: bobheadxi/deployments@v1.5.0
         id: deployment
-        if: ${{ !startsWith(github.ref, 'refs/heads/stable') }}
+        if: ${{ steps.decide_env.outputs.should_deploy }}
         with:
           step: start
           token: ${{ secrets.GITHUB_TOKEN }}
           env: ${{ steps.decide_env.outputs.env }}
 
       - name: deploy server
-        if: ${{ !startsWith(github.ref, 'refs/heads/stable') }}
+        if: ${{ steps.decide_env.outputs.should_deploy }}
         run: python tools/deploy_server.py
 
       - name: update deployment status
         uses: bobheadxi/deployments@v1.5.0
-        if: always() && !startsWith(github.ref, 'refs/heads/stable')
+        if: always() && steps.decide_env.outputs.should_deploy
```

Comment:
whats this always() doing

Reply:
Still run the step if the previous steps fails.